### PR TITLE
[Feature](mluOpExecFFT): fix accuracy and performance bug

### PR DESCRIPTION
**Pull request by aokbok**
Thanks for your contribution and we appreciate it a lot. :rocket::rocket:

## 1. Motivation

Please describe your motivation and the goal you want to achieve through this pull request.

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

Are new test cases added? If so, please post the corresponding generator-PR link here.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS™ Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- static threshold
  - diff1
    - [ ] float32 mlu diff1 <= 1e-5
    - [ ] float32 mlu diff1 <= 3e-3
    - [ ] float16 mlu diff1 <= 3e-3
  - diff2
    - [ ] float32 mlu diff2 <= 1e-5
    - [ ] float32 mlu diff2 <= 3e-3
    - [ ] float16 mlu diff2 <= 3e-3
  - diff3
    - [ ] mlu diff3 == 0
    - [ ] mlu diff3_1 == 0
    - [ ] mlu diff3_2 == 0
- dynamic threshold
  - [ ] diff1: mlu diff1 <= max(baseline diff1 * 10, static threshold)
  - [ ] diff2: mlu diff2 <= max(baseline diff2 * 10, static threshold)
  - [ ] diff3: mlu diff3 <= max(baseline diff3 * 10, static threshold)
    - float32, threshold = 1e-5
    - float16, threshold = 1e-3

#### 3.1.2 Operator Scheme checklist

- Supported hardware
  - [ ] MLU370
  - [ ] MLU590
- Job types
  - [ ] BLOCK
  - [ ] UNION1
  - [ ] UNION2
  - [ ] UNION4
  - [ ] The operator will dynamically select the most suitable task type, for example, UNION8

#### 3.2.1 Accuracy Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test (e.g. float32/int8)
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test, see: [Gencase-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/Gencase-User-Guide-zh.md)
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS™-Performance-Acceptance-Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md)

#### 3.2.2 Parameter Check

Test Point-1: `When a new operator is submitted, the test points are given and the test results are stated`. Acceptance Standard: `Normal error`.
```bash
Please fill your test results(Error Message) in here, ...
```

Test Point-2: `Whether illegal parameters are passed`. Acceptance Standard: `Normal error`.
```bash
Test results...
```

### 3.3 Performance Test

See [MLU-OPS™ Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

```bash
# The test results should contain Op name, Shape, Data type,  
#   MLU Hardware Time(us), MLU Interface Time(us), MLU IO Efficiency, 
#   MLU Compute Efficiency, and Mlu Workspace Size(Bytes)
# 
# for example:
#
# ----------- case0 -----------
# case0
# [Op name                ]: abs
# [Shape                  ]: input.shape=[1024,1024,3,4], output.shape=[1024,1024,3,4]
# [Data type]             ]: float32
# [MLU Hardware Time      ]: 15728 (us)
# [MLU Interface Time     ]: 369.008 (us)
# [MLU IO Efficiency      ]: 0.23275
# [MLU Compute Efficiency ]: 0.5
# [Mlu Workspace Size     ]: -1 (Bytes)
# 
# ----------- case1 -----------
# ...
```

Platform：MLU590
```bash
# ----------- case0 -----------
# ----------- case1 -----------
# ...
```

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

性能流水：
http://jenkins.svc.cambricon.com/dist/job/MLUOPS/view/Pipelines-Adaptable/job/MLU_ALL_PLATFORMS/job/mluops_build_perf_adaptable_mlu/482/
http://jenkins.svc.cambricon.com/dist/job/MLUOPS/view/Pipelines-Adaptable/job/MLU_ALL_PLATFORMS/job/mluops_build_perf_adaptable_mlu/481/
功能流水：
http://jenkins.svc.cambricon.com/dist/job/MLUOPS/job/MLU_ALL_PLATFORMS/job/mluops_build_test_adaptable_mlu/584/
http://jenkins.svc.cambricon.com/dist/job/MLUOPS/job/MLU_ALL_PLATFORMS/job/mluops_build_test_adaptable_mlu/585/
http://jenkins.svc.cambricon.com/dist/job/MLUOPS/job/MLU_ALL_PLATFORMS/job/mluops_build_test_adaptable_mlu/586/
http://jenkins.svc.cambricon.com/dist/job/MLUOPS/job/MLU_ALL_PLATFORMS/job/mluops_build_test_adaptable_mlu/587/

### CODE REVIEW

File: mlu_op.h [hunk 1]
```diff
@@ -14374,6 +14370,10 @@ mluOpSetFFTReserveArea(mluOpHandle_t handle, mluOpFFTPlan_t fft_plan, void *rese
  * when planning to use FFT with half-precision floating-point data, as it limits the flexibility compared to float data
  * types.
  *
+ * - For FFT 2D:
+ *     - real-to-complex FFT: output numbers / 2 + 1 should not be less than input numbers.
+ *     - complex-to-complex FFT: output numbers should not be less than input numbers.
+ *     - complex-to-real FFTs: output numbers should not be less than input numbers / 2 + 1.
```
Comment (AndyQiao0828) [inline mlu_op.h:14376]:
```suggestion
 *     - real-to-complex FFT: Output numbers / 2 + 1 should not be less than input numbers.
 *     - complex-to-complex FFT: Output numbers should not be less than input numbers.
 *     - complex-to-real FFT: Output numbers should not be less than input numbers / 2 + 1.
```
?

**Review by PetrelYy — APPROVED** [review]

**Review by PetrelYy — APPROVED** [review]

File: mlu_op.h (hunk 1 above)
Comment (aokbok) [reply mlu_op.h]: 好的，已修改

File: kernels/utils/cnnl_helper.cpp [hunk 2]
```diff
@@ -105,6 +105,71 @@ cnnlStatus_t mluOpConvertDescriptor(mluOpTensorDescriptor_t desc,
   return CNNL_STATUS_SUCCESS;
 }
 
+cnnlStatus_t mluOpConvertDescriptor_v2(mluOpTensorDescriptor_t desc,
+                                    cnnlTensorDescriptor_t _desc) {
```
Comment (zhengleiZL) [inline kernels/utils/cnnl_helper.cpp:109]: 对齐

File: kernels/utils/cnnl_helper.cpp [hunk 3]
```diff
@@ -105,6 +105,71 @@ cnnlStatus_t mluOpConvertDescriptor(mluOpTensorDescriptor_t desc,
...
+      MLUOP_STATUS_SUCCESS, "MLUOPS get tensor descriptor failed.",
+      CNNL_STATUS_INTERNAL_ERROR);
+  CHECK_FUNC_RETURN(mluOpGetTensorDescriptorOnchipDataType(desc, &onchip_dtype),
+                    MLUOP_STATUS_SUCCESS,
+                    "MLUOPS get tensor descriptor onchip type failed.",
+                    CNNL_STATUS_INTERNAL_ERROR);
+  int64_t *dims = new int64_t[tensor_dim];
+  int64_t *strides = new int64_t[tensor_dim];
+  CHECK_FUNC_RETURN(mluOpGetTensorDescriptorEx_v2(desc, &layout, &dtype,
+                                               &tensor_dim, dims, strides),
```
Comment (zhengleiZL) [inline kernels/utils/cnnl_helper.cpp:127]: 同上

**Comment by aokbok** [comment]
性能数据：
image

**Review by zhengleiZL — APPROVED** [review]

**Review by AndyQiao0828 — APPROVED** [review]

**Pull request merged**